python/plugins/processing/algs/gdal/GdalUtils.py

```diff
@@ -210,20 +210,24 @@ def getSupportedOutputRasters():
 
     @staticmethod
     def getSupportedRasterExtensions():
-        allexts = ['tif']
+        allexts = []
         for exts in list(GdalUtils.getSupportedRasters().values()):
             for ext in exts:
-                if ext not in allexts and ext != '':
+                if ext not in allexts and ext not in ['', 'tif', 'tiff']:
                     allexts.append(ext)
+        allexts.sort()
+        allexts[0:0] = ['tif', 'tiff']
         return allexts
 
     @staticmethod
     def getSupportedOutputRasterExtensions():
-        allexts = ['tif']
+        allexts = []
         for exts in list(GdalUtils.getSupportedOutputRasters().values()):
             for ext in exts:
-                if ext not in allexts and ext != '':
+                if ext not in allexts and ext not in ['', 'tif', 'tiff']:
```

Comment:
Having a format appear first while everything else is sorted will lead to confusion. I suggest to either leave everything sorted or, if you insist on tiff being first, let it appear twice, once at the beginning and once at its sorted position.

Reply:
I think having the most used formats easily accessible is a sensible choice.

Reply:
I have no strong preference for sorting the list. I kept the previous behaviour regarding the TIF on top because I thought it was set for a design choice that I didn't intend to change.

So I could remove the last commit and leave the list unsorted with TIF on top like before (this was only like a cosmetic feature added to the bug fix).

Reply:
Let's defer this for a separate discussion. Right now this pr follows the approach used elsewhere in qgis spatial data file pickers where tiff/gpkg/shp are always listed first and once only. If we change this, we should do it everywhere at once.